### ISSUE TITLE
File-Based Catalog/FBC does not honor --use-http or --skip-tls-verify

### ISSUE THREAD

**Issue by loganmc10** [operator-framework/operator-sdk#5910]
When using ```operator-sdk run bundle --skip-tls-verify```:

> msg="Failed to run bundle: error generating File-Based Catalog with bundle \"localhost/routernetes-operator-bundle:v0.0.1\": error creating a File-Based Catalog with image \"localhost/routernetes-operator-bundle:v0.0.1\": error in rendering the bundle and index image: render reference \"localhost/routernetes-operator-bundle:v0.0.1\": error resolving name : failed to do request: Head \"https://localhost/v2/routernetes-operator-bundle/manifests/v0.0.1\": x509: certificate signed by unknown authority\n"

When using ```operator-sdk run bundle --use-http```:

> msg="Failed to run bundle: error generating File-Based Catalog with bundle \"localhost:5000/routernetes-operator-bundle:v0.0.1\": error creating a File-Based Catalog with image \"localhost:5000/routernetes-operator-bundle:v0.0.1\": error in rendering the bundle and index image: render reference \"localhost:5000/routernetes-operator-bundle:v0.0.1\": error resolving name : failed to do request: Head \"https://localhost:5000/v2/routernetes-operator-bundle/manifests/v0.0.1\": http: server gave HTTP response to HTTPS client\n"

**Comment by laxmikantbpandhare** [comment]
I tried `--skip-tls-verify` and `--use-http` with `memcached-operator` and I am able to run it successfully with `operator-sdk run bundle`:

```
laxmikantbhaskarpandhare@lpandhar-mac memcached-quarkus-operator % operator-sdk run bundle docker.io/013859989/memcached-quarkus-operator-bundle:v0.0.1 --timeout=10m0s --skip-tls-verify
INFO[0007] Creating a File-Based Catalog of the bundle "docker.io/013859989/memcached-quarkus-operator-bundle:v0.0.1" 
INFO[0009] Generated a valid File-Based Catalog         
INFO[0010] Created registry pod: docker-io-013859989-memcached-quarkus-operator-bundle-v0-0-1 
INFO[0010] Created CatalogSource: memcached-quarkus-operator-catalog 
INFO[0010] OperatorGroup "operator-sdk-og" created      
INFO[0010] Created Subscription: memcached-quarkus-operator-v0-0-1-sub 
INFO[0012] Approved InstallPlan install-hnjnm for the Subscription: memcached-quarkus-operator-v0-0-1-sub 
INFO[0012] Waiting for ClusterServiceVersion "default/memcached-quarkus-operator.v0.0.1" to reach 'Succeeded' phase 
INFO[0012]   Waiting for ClusterServiceVersion "default/memcached-quarkus-operator.v0.0.1" to appear 
INFO[0021]   Found ClusterServiceVersion "default/memcached-quarkus-operator.v0.0.1" phase: Installing 
INFO[0052]   Found ClusterServiceVersion "default/memcached-quarkus-operator.v0.0.1" phase: Succeeded 
INFO[0052] OLM has successfully installed "memcached-quarkus-operator.v0.0.1"
```

Could you please provide more details about the `operator-sdk` version and the exact command that you are trying to run using `operator-sdk run bundle`

**Comment by loganmc10** [comment]
I'm using Operator SDK 1.22.0. I assume your example works because the image is hosted on docker.io. I am testing using a local registry (https://docs.docker.com/registry/deploying/), which can either be served using only HTTP, or HTTPS with a self-signed certificate.

The command I am doing is:
```
operator-sdk run bundle ${IMAGE_TAG_BASE}-bundle:v0.0.0 -n operators --skip-tls-verify --timeout 10m
```

I am doing this as part of a PR workload (when a PR I submitted, GitHub Actions builds the operator, uploads it to a local registry, then installs the operator from that local registry for testing)

**Comment by laxmikantbpandhare** [comment]
Ah! Let me try with the local registry. I will update you here soon.

**Comment by jmrodri** [comment]
> possibly a release blocker?

**Comment by laxmikantbpandhare** [comment]
@loganmc10 - This issue is fixed and we will release the Operator SDK 1.22.1 patch release in the next couple of days. Once it got released, please confirm the above-mentioned issue.

**Comment by loganmc10** [comment]
I worked around this by adding the self-signed certificate for the local registry to the certificate store on the CI server, I assume it's good now, although I don't see anything in the 1.22.1 changelog/release notes that mentions this being fixed

**Comment by rhyas** [comment]
FWIW, this bug does not appear to be fixed.

```

**Comment by rhyas** [comment]
Ok, this appears to just have not completely made it into 1.22.1 as the thread suggests. Building the project off the main branch results in a build that does accept and use the flags as it should, so I suspect it'll be in a future release. Just a suggestion, but you should tag this issue with the release that it finally makes it into. It will save some time if people come across this.

**Comment by laxmikantbpandhare** [comment]
@rhyas - Yes, these changes are not get added in the 1.22.1 patch release. We are planning to do 1.23.0 in the next couple of days. We will make sure that these changes are part of the upcoming release.

**Comment by laxmikantbpandhare** [comment]
We released SDK v1.22.2, changes for `skip-tls-verify` and `use-http` is added as part of this release.
